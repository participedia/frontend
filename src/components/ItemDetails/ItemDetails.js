import React from "react";
import auth from "../../utils/AuthService";
import { FormattedMessage } from "react-intl";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { Container, Row, Col } from "reactstrap";
import { FormattedDate } from "react-intl";
import Gallery from "../Gallery";
import BookmarkToggle from "../BookmarkToggle";
import AccordionTab from "../AccordionTab/AccordionTab";
import FloatingActionButton from "material-ui/FloatingActionButton";
import ContentPencil from "material-ui/svg-icons/image/edit";
import caseIconFB from "../../img/pp-case-icon-fb.svg";
import caseIconTW from "../../img/pp-case-icon-tw.svg";
import caseIconLN from "../../img/pp-case-icon-ln.png";
import { ShareButtons } from "react-share";
import htmlToText from "html-to-text";
import "./ItemDetails.css";
import Toggle from "material-ui/Toggle";

function isCurator(profile) {
  if (!profile || !profile.app_metadata || !profile.app_metadata.authorization)
    return false;
  let groups = profile.app_metadata.authorization.groups;
  return groups.indexOf("Curators") !== -1;
}

class Featured extends React.Component {
  static propTypes = {
    toggleFeatured: PropTypes.func.isRequired,
    thing: PropTypes.object.isRequired
  };

  onToggle(object, value) {
    this.props.thing["featured"] = value;
    this.props.toggleFeatured(this.props.thing, value);
  }

  render() {
    const { isAuthenticated } = auth;
    const { profile } = this.props;
    if (!isAuthenticated) return <div />;
    return isCurator(profile) ? (
      <div className="featuretoggle">
        <Toggle
          onToggle={this.onToggle.bind(this)}
          labelPosition="right"
          defaultToggled={this.props.thing.featured}
          label="Featured"
        />
      </div>
    ) : (
      <div />
    );
  }
}

class Hidden extends React.Component {
  static propTypes = {
    toggleHidden: PropTypes.func.isRequired,
    thing: PropTypes.object.isRequired
  };

  onToggle(object, value) {
    this.props.thing["hidden"] = value;
    this.props.toggleHidden(this.props.thing, value);
  }

  render() {
    const { isAuthenticated, profile } = this.props;
    if (!isAuthenticated) return <div />;
    return isCurator(profile) ? (
      <div className="featuretoggle">
        <Toggle
          onToggle={this.onToggle.bind(this)}
          labelPosition="right"
          defaultToggled={this.props.thing.hidden}
          label="Hidden"
        />
      </div>
    ) : (
      <div />
    );
  }
}

const defaultThing = {
  title: "Loading",
  type: "content",
  body: "Description",
  images: [],
  post_date: "2011-11-30T17:20:28",
  updated_date: "2011-11-30T17:20:28",
  location: {
    city: "",
    countrycode: ""
  },
  videos: [],
  creator: { name: "author name" },
  last_updated_by: { name: "author name" }
};

export default class ItemDetails extends React.Component {
  componentWillMount() {
    this.setState({ profile: {} });
    const { isAuthenticated, userProfile, getProfile } = auth;
    if (!userProfile) {
      getProfile((err, profile) => {
        this.setState({ profile, isAuthenticated: isAuthenticated() });
      });
    } else {
      this.setState({
        profile: userProfile,
        isAuthenticated: isAuthenticated()
      });
    }
  }

  render() {
    const {
      FacebookShareButton,
      TwitterShareButton,
      LinkedinShareButton
    } = ShareButtons;
    const intl = this.props.intl;
    const thing = this.props.data || defaultThing;
    let bookmarked = thing.bookmarked;
    let bookmarkIcon = (
      <BookmarkToggle
        thingType={thing.type}
        thingid={thing.id}
        bookmarked={bookmarked}
      />
    );
    let bodyText = htmlToText.fromString(thing.body);
    let currentUrl =
      process.env.REACT_APP_ROOT_URL + this.props.location.pathname;
    let textFacebook = bodyText.substring(0, 240) + "...";
    let creator = thing.creator;
    let creator_url = "/users/" + creator.user_id;
    let creator_name = creator.name;
    let last_updated_by = thing.last_updated_by;
    let last_updated_by_name = last_updated_by.name;
    let last_updated_by_url = "/users/" + last_updated_by.user_id;
    let id = this.props.id;
    let type = thing.type;
    let editLinkUrl = `/${type}/${id}/edit`;
    let detailedBits = React.createElement(this.props.details, {
      case: thing,
      intl
    });

    // XXX L10N -- lots of strings to extract.
    return (
      <div>
        <div className="main-contents">
          <Container className="detailed-case-component" fluid>
            <Row>
              <Col
                md="3"
                className="d-none d-sm-block d-md-block d-lg-block d-xl-block sidepanel"
              >
                <Featured
                  thing={thing}
                  profile={this.state.profile}
                  isAuthenticated={this.state.isAuthenticated}
                  toggleFeatured={this.props.toggleFeatured}
                />
                <Hidden
                  thing={thing}
                  profile={this.state.profile}
                  isAuthenticated={this.state.isAuthenticated}
                  toggleHidden={this.props.toggleHidden}
                />
                {detailedBits}
              </Col>
              <Col md="6" xs="12" className="main-area">
                <div className="case-box">
                  <h3 className="category">
                    <FormattedMessage id={thing.type} />
                  </h3>
                  <h1 className="case-title">{thing.title}</h1>
                  <ul className="icons-mobile clearfix d-md-none d-lg-none d-xl-none">
                    <li>{bookmarkIcon}</li>
                    <li>
                      <FacebookShareButton
                        url={currentUrl}
                        quote={textFacebook}
                      >
                        <img src={caseIconFB} alt="" />
                      </FacebookShareButton>
                    </li>
                    <li>
                      <TwitterShareButton url={currentUrl} title={thing.title}>
                        <img src={caseIconTW} alt="" />
                      </TwitterShareButton>
                    </li>
                    <li>
                      <LinkedinShareButton
                        url={currentUrl}
                        description={textFacebook}
                        title={thing.title}
                      >
                        <img src={caseIconLN} alt="" />
                      </LinkedinShareButton>
                    </li>
                  </ul>
                  <Gallery thing={thing} />
                  <div className="mobile-metadata accordion d-md-none d-lg-none d-xl-none">
                    <AccordionTab titleId={thing.type + "_data"}>
                      <div className="content">{detailedBits}</div>
                    </AccordionTab>
                  </div>
                  <div className="authorship-details">
                    <p className="author-line">
                      First submitted by&nbsp;
                      <Link to={creator_url}>{creator_name}</Link>
                    </p>
                    <p className="date-line">
                      {thing.post_date ? (
                        <FormattedDate
                          value={thing.post_date}
                          year="numeric"
                          month="long"
                          day="2-digit"
                        />
                      ) : (
                        <span />
                      )}
                    </p>
                    <p className="author-line">
                      Most recent changes by&nbsp;
                      <Link to={last_updated_by_url}>
                        {last_updated_by_name}
                      </Link>
                    </p>
                    <p className="date-line">
                      {thing.updated_date ? (
                        <FormattedDate
                          value={thing.updated_date}
                          year="numeric"
                          month="long"
                          day="2-digit"
                        />
                      ) : (
                        <span />
                      )}
                    </p>
                  </div>
                  <div
                    className="case-html"
                    dangerouslySetInnerHTML={{ __html: thing.body }}
                  />
                </div>
              </Col>
              <Col
                md="3"
                className="case-tools d-none d-sm-block d-md-block d-lg-block d-xl-block"
              >
                <div className="top-icons">
                  {bookmarkIcon}
                  <FacebookShareButton url={currentUrl} quote={textFacebook}>
                    <img src={caseIconFB} alt="" />
                  </FacebookShareButton>
                  <TwitterShareButton url={currentUrl} title={thing.title}>
                    <img src={caseIconTW} alt="" />
                  </TwitterShareButton>
                  <LinkedinShareButton
                    url={currentUrl}
                    description={textFacebook}
                    title={thing.title}
                  >
                    <img src={caseIconLN} alt="" />
                  </LinkedinShareButton>
                </div>
              </Col>
            </Row>
          </Container>
          <Link to={editLinkUrl}>
            <FloatingActionButton secondary={true} className="editButton">
              <ContentPencil />
            </FloatingActionButton>
          </Link>
        </div>
      </div>
    );
  }
}
