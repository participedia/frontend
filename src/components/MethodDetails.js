import React from "react";
import { FormattedMessage } from "react-intl";
import LinkedPropertyGroupWithHeading from "./LinkedPropertyGroupWithHeading";
import ListWithHeading from "./ListWithHeading";
import TextListWithHeading from "./TextListWithHeading";
import { BooleanProp, TextProp } from "../components/Props";

export default class MethodDetails extends React.Component {
  render() {
    let thing = this.props.case; // should thing be `this.props.method`?
    return (
      <div>
        <h2 className="d-none d-sm-block d-md-block d-lg-block d-xl-block sub-heading">
          <FormattedMessage id="data" />
        </h2>

        <ListWithHeading heading="links" property="links" thing={thing} />

        <BooleanProp
          heading="facilitated"
          label="facilitated"
          property="facilitated"
          thing={thing}
        />

        <TextListWithHeading property="typical_purposes" thing={thing} />

        <TextProp
          label="participant_selection"
          property="participant_selection"
          thing={thing}
        />

        <TextProp
          label="recruitment_method"
          property="recruitment_method"
          thing={thing}
        />

        <ListWithHeading
          heading="interaction_modes"
          property="interaction_modes"
          thing={thing}
        />

        <LinkedPropertyGroupWithHeading
          heading="communication_outcomes"
          property="communication_outcomes"
          thing={thing}
        />

        <LinkedPropertyGroupWithHeading
          heading="decision_method"
          property="decision_method"
          thing={thing}
        />

        <LinkedPropertyGroupWithHeading
          heading="geographical_scope"
          property="geographical_scope"
          thing={thing}
        />

        <LinkedPropertyGroupWithHeading
          heading="issue_polarization"
          property="issue_polarization"
          thing={thing}
        />

        <LinkedPropertyGroupWithHeading
          heading="issue_technical_complexity"
          property="issue_technical_complexity"
          thing={thing}
        />

        <LinkedPropertyGroupWithHeading
          heading="issue_interdependency"
          property="issue_interdependency"
          thing={thing}
        />
      </div>
    );
  }
}
