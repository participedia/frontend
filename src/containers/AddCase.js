import CaseForm from '../components/CaseForm'
import { reduxForm } from 'redux-form'

const mapStateToProps = (state /* , ownProps */) => {
  return state.form
}

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    onSubmit: (data) => {
    }
  }
}

export const fields = [ 'firstName', 'lastName', 'age', 'color', 'bio' ]

let AddCaseForm = reduxForm({
  form: 'ncase',
  fields: ['firstName', 'lastName', 'email']
}, mapStateToProps, mapDispatchToProps)(CaseForm)

export default AddCaseForm
