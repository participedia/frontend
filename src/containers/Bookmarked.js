import React from "react";
import api from "../utils/api";
import SearchResults from "./SearchResults";

export default class Bookmarked extends React.Component {
  constructor(props) {
    super(props);
    this.state = { data: null };
  }
  componentDidMount(props) {
    api.getBookmarkedItems().then(data => this.setState({ data }));
  }
  render() {
    if (this.state.data) {
      return <SearchResults data={this.state.data} />;
    } else {
      return <div />;
    }
  }
}
