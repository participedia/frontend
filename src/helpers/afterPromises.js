// shamelessly borrowed from https://github.com/facebook/jest/issues/2136#issuecomment-266817172

export default function afterPromises(done, fn) {
  setTimeout(
    function() {
      try {
        fn();
        done();
      } catch (e) {
        done.fail(e);
      }
    },
    0
  );
}
