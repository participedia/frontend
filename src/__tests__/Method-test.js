import React from "react";
import renderer from "react-test-renderer";

import MethodDetails from "../components/MethodDetails";
import MethodEditor from "../components/MethodEditor";
import ItemDetails from "../components/ItemDetails/ItemDetails";
import { IntlProvider } from "react-intl";
import { getBestMatchingMessages } from "../utils/l10n";

let locale = "en-US";
let messages = getBestMatchingMessages(locale);
const intlProvider = new IntlProvider({ locale: locale }, {});
const { intl } = intlProvider.getChildContext();

import { MemoryRouter } from "react-router";
import { Method } from "../containers/Method";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import getMuiTheme from "material-ui/styles/getMuiTheme";
const muiTheme = getMuiTheme({});
import { mountWithIntl } from "../helpers/intl-enzyme-test-helper.js";
import data from "./method_data.json";
import intlProps from "../helpers/intl-props-test-helper.js";
import afterPromises from "../helpers/afterPromises";
import injectTapEventPlugin from "react-tap-event-plugin";
injectTapEventPlugin();

let fetchMock = require("fetch-mock");

jest.mock("react-timeago", () => {
  const React = require("react");
  const TimeAgo = props => React.createElement("TimeAgo");
  return TimeAgo;
}); // so we don't break tests as time goes by

fetchMock.get(
  process.env.REACT_APP_API_URL + "/method/145",
  JSON.stringify({ data: data })
);

jest.mock("material-ui/FloatingActionButton");
jest.mock("../components/BookmarkToggle");
jest.mock("material-ui/TextField", () => "Textfield");
jest.mock("material-ui/RaisedButton");
jest.mock("material-ui/RadioButton");
jest.mock("material-ui/DatePicker");
jest.mock("material-ui/SelectField");
jest.mock("material-ui/SvgIcon");
jest.mock("react-geosuggest");
jest.mock("material-ui-chip-input");
jest.mock("../vendor/react-items-list");
jest.mock("../components/BodyEditor");

function setup() {
  const props = {
    intl: intlProps,
    auth: { getProfile: cb => cb({}), isAuthenticated: () => false },
    location: { pathname: "/method/145" },
    match: { params: { nodeID: 145 } },
    toggleFeatured: function() {},
    handleInternal: function() {},
    data: data
  };

  const enzymeWrapper = mountWithIntl(
    <IntlProvider locale={locale} messages={messages}>
      <MemoryRouter>
        <MuiThemeProvider muiTheme={muiTheme}>
          <Method {...props} intl={intl} />
        </MuiThemeProvider>
      </MemoryRouter>
    </IntlProvider>
  );
  return {
    props,
    enzymeWrapper
  };
}

describe("containers", () => {
  describe("Method", () => {
    it("should render proper data for method", done => {
      const { enzymeWrapper } = setup();
      afterPromises(done, () => {
        expect(enzymeWrapper.find("p.author-line").length).toBe(2);
        expect(enzymeWrapper.find("h1.case-title").text()).not.toBe("");
      });
    });
  });
});

let props = setup().props;
props["details"] = MethodDetails;

test("ItemDetails for Method renders correctly", () => {
  const tree = renderer
    .create(
      <IntlProvider locale={locale} messages={messages}>
        <MemoryRouter>
          <ItemDetails {...props} toggleHidden={function() {}} intl={intl} />
        </MemoryRouter>
      </IntlProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Method editor renders correctly", () => {
  const tree = renderer
    .create(
      <IntlProvider locale={locale} messages={messages}>
        <MemoryRouter>
          <MuiThemeProvider muiTheme={muiTheme}>
            <MethodEditor {...props} type="method" thing={data} intl={intl} />
          </MuiThemeProvider>
        </MemoryRouter>
      </IntlProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
