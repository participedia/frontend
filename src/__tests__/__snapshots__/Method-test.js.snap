// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetails for Method renders correctly 1`] = `
<div>
  <div
    className="main-contents"
  >
    <div
      className="detailed-case-component container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="d-none d-sm-block d-md-block d-lg-block d-xl-block sidepanel col col-md-3"
        >
          <div />
          <div />
          <div />
          <div>
            <h2
              className="d-none d-sm-block d-md-block d-lg-block d-xl-block sub-heading"
            >
              <span>
                Data
              </span>
            </h2>
            <div />
            <div />
            <div>
              <p
                className="sub-sub-heading"
              >
                <span>
                  Relevant Links (URLs)
                </span>
              </p>
              <div
                className="links"
              >
                <div>
                  <a
                    href="http://link 1"
                    target="_blank"
                  >
                    link 1
                  </a>
                </div>
                <div>
                  <a
                    href="http://link 2"
                    target="_blank"
                  >
                    link 2
                  </a>
                </div>
                <div>
                  <a
                    href="http://link 3"
                    target="_blank"
                  >
                    link 3
                  </a>
                </div>
              </div>
            </div>
            <div />
            <div />
            <div />
            <div />
            <div />
            <div
              className="linked-property"
            >
              <p
                className="sub-sub-heading"
              >
                <span>
                  Facilitated
                </span>
              </p>
              <div
                className="facilitated"
              >
                <span>
                  False
                </span>
              </div>
            </div>
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
            <div />
          </div>
        </div>
        <div
          className="main-area col-12 col-md-6"
        >
          <div
            className="case-box"
          >
            <h3
              className="category"
            >
              <span>
                Method
              </span>
            </h3>
            <h1
              className="case-title"
            >
              Petition
            </h1>
            <ul
              className="icons-mobile clearfix d-md-none d-lg-none d-xl-none"
            >
              <li />
              <li>
                <div
                  className="SocialMediaShareButton SocialMediaShareButton--facebook"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  style={Object {}}
                  tabIndex="0"
                >
                  <img
                    alt=""
                    src="pp-case-icon-fb.svg"
                  />
                </div>
              </li>
              <li>
                <div
                  className="SocialMediaShareButton SocialMediaShareButton--twitter"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  style={Object {}}
                  tabIndex="0"
                >
                  <img
                    alt=""
                    src="pp-case-icon-tw.svg"
                  />
                </div>
              </li>
              <li>
                <div
                  className="SocialMediaShareButton SocialMediaShareButton--linkedin"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  style={Object {}}
                  tabIndex="0"
                >
                  <img
                    alt=""
                    src="pp-case-icon-ln.png"
                  />
                </div>
              </li>
            </ul>
            <div />
            <div
              className="mobile-metadata accordion d-md-none d-lg-none d-xl-none"
            >
              <div
                className="tab"
                onClick={[Function]}
              >
                <h3
                  className="title"
                >
                  <span>
                    Method Data
                  </span>
                </h3>
                <div
                  className="content"
                >
                  <div>
                    <h2
                      className="d-none d-sm-block d-md-block d-lg-block d-xl-block sub-heading"
                    >
                      <span>
                        Data
                      </span>
                    </h2>
                    <div />
                    <div />
                    <div>
                      <p
                        className="sub-sub-heading"
                      >
                        <span>
                          Relevant Links (URLs)
                        </span>
                      </p>
                      <div
                        className="links"
                      >
                        <div>
                          <a
                            href="http://link 1"
                            target="_blank"
                          >
                            link 1
                          </a>
                        </div>
                        <div>
                          <a
                            href="http://link 2"
                            target="_blank"
                          >
                            link 2
                          </a>
                        </div>
                        <div>
                          <a
                            href="http://link 3"
                            target="_blank"
                          >
                            link 3
                          </a>
                        </div>
                      </div>
                    </div>
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div
                      className="linked-property"
                    >
                      <p
                        className="sub-sub-heading"
                      >
                        <span>
                          Facilitated
                        </span>
                      </p>
                      <div
                        className="facilitated"
                      >
                        <span>
                          False
                        </span>
                      </div>
                    </div>
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                  </div>
                </div>
              </div>
            </div>
            <div
              className="authorship-details"
            >
              <p
                className="author-line"
              >
                First submitted by 
                <a
                  href="/users/undefined"
                  onClick={[Function]}
                >
                  Gsmith
                </a>
              </p>
              <p
                className="date-line"
              >
                <span>
                  November 30, 2011
                </span>
              </p>
              <p
                className="author-line"
              >
                Most recent changes by 
                <a
                  href="/users/undefined"
                  onClick={[Function]}
                >
                  Gsmith
                </a>
              </p>
              <p
                className="date-line"
              >
                <span>
                  December 01, 2011
                </span>
              </p>
            </div>
            <div
              className="case-html"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "**Definition**
                
                **Problems and Purpose**
                
                **History**
                
                **Participant Selection**
                
                **Deliberation, Decisions, and Public Interaction**
                
                **Influence, Outcomes, and Effects**
                
                **Analysis and Criticism**
                
                **Secondary Sources**
                
                **External Links**
                
                **Notes**",
                }
              }
            />
          </div>
        </div>
        <div
          className="case-tools d-none d-sm-block d-md-block d-lg-block d-xl-block col col-md-3"
        >
          <div
            className="top-icons"
          >
            <div
              className="SocialMediaShareButton SocialMediaShareButton--facebook"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              style={Object {}}
              tabIndex="0"
            >
              <img
                alt=""
                src="pp-case-icon-fb.svg"
              />
            </div>
            <div
              className="SocialMediaShareButton SocialMediaShareButton--twitter"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              style={Object {}}
              tabIndex="0"
            >
              <img
                alt=""
                src="pp-case-icon-tw.svg"
              />
            </div>
            <div
              className="SocialMediaShareButton SocialMediaShareButton--linkedin"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              style={Object {}}
              tabIndex="0"
            >
              <img
                alt=""
                src="pp-case-icon-ln.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <a
      href="/method/undefined/edit"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`Method editor renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="main-contents"
  >
    <div
      className="detailed-case-component container-fluid"
    >
      <div
        className="d-none d-sm-block d-md-block d-lg-block d-xl-block sidepanel col col-md-3"
      />
      <div
        className="ml-auto mr-auto col col-md-6"
      >
        <div
          className="case-box"
        >
          <div
            className="field-case top"
          >
            <h2
              className="sub-heading"
            >
              <label
                htmlFor="title"
              >
                <span>
                  Title of Method
                </span>
              </label>
            </h2>
            <p
              className="explanatory-text"
            >
              <span>
                method_title_placeholder
              </span>
            </p>
            <Textfield
              className="custom-field"
              disabled={false}
              errorText={undefined}
              floatingLabelText=""
              fullWidth={true}
              hintText={null}
              name="title"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder=""
              type="text"
              value="Petition"
            />
          </div>
          <div
            className="field-case"
          >
            <h2
              className="sub-heading"
            >
              <span>
                Relevant Links (URLs)
              </span>
            </h2>
            <div>
              <h3
                className="sub-heading"
              >
                Relevant Links (URLs)
              </h3>
              <p
                className="explanatory-text"
              >
                <span>
                  If there is a main website for this case, enter it here. Add links to additional sources so that readers and editors can find more information about this case online. 
                </span>
              </p>
              <div
                className="list links"
              />
            </div>
          </div>
          <div
            className="field-case media"
          >
            <h2
              className="sub-heading"
            >
              <span>
                Media
              </span>
            </h2>
            <h3
              className="sub-sub-heading"
            >
              <span>
                Photos
              </span>
            </h3>
            <div>
              <div
                className="itemPics row"
              />
              <p
                className="explanatory-text"
              >
                <span>
                  Select a photo from your device.
                </span>
              </p>
              <div>
                <span>
                  Sorry, upload only works if you're logged in.
                </span>
              </div>
            </div>
            <h3
              className="sub-sub-heading pt-4"
            >
              <span>
                Videos
              </span>
            </h3>
            <div>
              <h3
                className="sub-heading"
              >
                Videos
              </h3>
              <p
                className="explanatory-text"
              >
                <span>
                  Add a video link (Vimeo or YouTube)
                </span>
              </p>
              <div
                className="list videos"
              />
            </div>
          </div>
          <div
            className="field-case"
          >
            <h2
              className="sub-heading"
            >
              <span>
                Tags
              </span>
            </h2>
            <div>
              <h3
                className="sub-heading"
              >
                <span>
                  Tags
                </span>
              </h3>
              <p
                className="explanatory-text"
              >
                <span>
                  Select tags that best describe this case.
                </span>
              </p>
            </div>
          </div>
        </div>
        <div>
          <div>
            <div
              className="field-case select"
            >
              <h3
                className="sub-heading"
              >
                General Issues
              </h3>
              <p
                className="explanatory-text"
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Select an issue",
                    }
                  }
                />
              </p>
            </div>
            <div
              className="field-case"
            >
              <h2
                className="sub-heading"
                htmlFor="body_en"
              >
                Method description
              </h2>
            </div>
            <div
              className="field-case select"
            >
              <h3
                className="sub-heading"
              >
                Kind of influence
              </h3>
              <p
                className="explanatory-text"
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Select kind of influence",
                    }
                  }
                />
              </p>
            </div>
            <div
              className="field-case select"
            >
              <h3
                className="sub-heading"
              >
                Communication Mode
              </h3>
              <p
                className="explanatory-text"
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Select a communication mode",
                    }
                  }
                />
              </p>
            </div>
            <div
              className="field-case select"
            >
              <h3
                className="sub-heading"
              >
                Communication with audience
              </h3>
              <p
                className="explanatory-text"
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Select a communication with audience",
                    }
                  }
                />
              </p>
            </div>
            <div
              className="field-case select"
            >
              <h3
                className="sub-heading"
              >
                Face-to-face, online or both
              </h3>
              <p
                className="explanatory-text"
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Did participants interact face-to-face, online, or both?",
                    }
                  }
                />
              </p>
            </div>
            <div
              className="field-case"
            >
              <h2
                className="sub-heading boolean"
              >
                <span>
                  Facilitated
                </span>
              </h2>
              <div
                className="facilitated"
              />
            </div>
            <div
              className="field-case select"
            >
              <h3
                className="sub-heading"
              >
                Best for
              </h3>
              <p
                className="explanatory-text"
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Select Best for",
                    }
                  }
                />
              </p>
            </div>
            <div
              className="field-case select"
            >
              <h3
                className="sub-heading"
              >
                Decision method
              </h3>
              <p
                className="explanatory-text"
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Select a decision method",
                    }
                  }
                />
              </p>
            </div>
            <div
              className="field-case select"
            >
              <h3
                className="sub-heading"
              >
                Governance contribution
              </h3>
              <p
                className="explanatory-text"
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Select Governance contribution",
                    }
                  }
                />
              </p>
            </div>
            <div
              className="field-case select"
            >
              <h3
                className="sub-heading"
              >
                Issue interdependency
              </h3>
              <p
                className="explanatory-text"
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Select Issue interdependency",
                    }
                  }
                />
              </p>
            </div>
            <div
              className="field-case select"
            >
              <h3
                className="sub-heading"
              >
                Issue polarization
              </h3>
              <p
                className="explanatory-text"
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Select Issue polarization",
                    }
                  }
                />
              </p>
            </div>
            <div
              className="field-case select"
            >
              <h3
                className="sub-heading"
              >
                Issue technical complexity
              </h3>
              <p
                className="explanatory-text"
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Select Issue technical complexity",
                    }
                  }
                />
              </p>
            </div>
            <div
              className="field-case select"
            >
              <h3
                className="sub-heading"
              >
                Public interaction method
              </h3>
              <p
                className="explanatory-text"
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Select Public interaction method",
                    }
                  }
                />
              </p>
            </div>
            <div
              className="field-case select"
            >
              <h3
                className="sub-heading"
              >
                Method of interaction
              </h3>
              <p
                className="explanatory-text"
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Select Method of interaction",
                    }
                  }
                />
              </p>
            </div>
            <div
              className="field-case select"
            >
              <h3
                className="sub-heading"
              >
                Typical funding source
              </h3>
              <p
                className="explanatory-text"
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Select Typical funding source",
                    }
                  }
                />
              </p>
            </div>
            <div
              className="field-case select"
            >
              <h3
                className="sub-heading"
              >
                Type of implementing entity
              </h3>
              <p
                className="explanatory-text"
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Select type of implementing entity",
                    }
                  }
                />
              </p>
            </div>
            <div
              className="field-case select"
            >
              <h3
                className="sub-heading"
              >
                Type of sponsoring entity
              </h3>
              <p
                className="explanatory-text"
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Select type of sponsoring entity",
                    }
                  }
                />
              </p>
            </div>
            <div
              className="field-case last related"
            >
              <h2
                className="sub-heading"
              >
                <span>
                  Related Content
                </span>
              </h2>
              <div
                className="related-content"
              >
                <h3
                  className="sub-sub-heading"
                >
                  <span>
                    Related Cases
                  </span>
                </h3>
                <div>
                  <h3
                    className="sub-heading"
                  >
                    <span>
                      Related Cases
                    </span>
                  </h3>
                  <p
                    className="explanatory-text"
                  >
                    <span>
                      Begin typing the title of an existing case
                    </span>
                  </p>
                </div>
                <h3
                  className="sub-sub-heading mt-3"
                >
                  <span>
                    Related Methods
                  </span>
                </h3>
                <div>
                  <h3
                    className="sub-heading"
                  >
                    <span>
                      Related Methods
                    </span>
                  </h3>
                  <p
                    className="explanatory-text"
                  >
                    <span>
                      Begin typing the title of an existing method
                    </span>
                  </p>
                </div>
                <h3
                  className="sub-sub-heading mt-3"
                >
                  <span>
                    Related Organizations
                  </span>
                </h3>
                <div>
                  <h3
                    className="sub-heading"
                  >
                    <span>
                      Related Organizations
                    </span>
                  </h3>
                  <p
                    className="explanatory-text"
                  >
                    <span>
                      Begin typing the title of an existing organization
                    </span>
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;
